### ISSUE TITLE
Integrate an updated version of rah_knots into the core.

### ISSUE THREAD

**Issue by GugUser** [textpattern/textpattern#924]
It’s not much code and would bee very useful. Thanks.

**Comment by makss** [comment]
It might be useful to add other extensions, such as keyboard TAB support. (upm_insert_tab or [some other](https://www.google.com/search?q=javascript+insert+tab+into+textarea&oq=javascript+insert+tab+into+textarea&gws_rd=cr) variant)

**Comment by philwareham** [comment]
> It might be useful to add other extensions, such as keyboard TAB support.

As long as it doesn't interfere with the standard tabbing through the UI, then sure.

**Comment by philwareham** [comment]
The new floating save button (see https://github.com/textpattern/textpattern/commit/f6d6a7b05183d412cff9397380514b0903be51e2) might negate the need for this. Test it out and let me know.

**Comment by phiw13** [comment]
> The new floating save button (see f6d6a7b) might negate the need for this

Not completely. There is another usability improvement provided by this extension: when using the keyboard shortcut, **the focus remains fully on the textarea** – IOW: edit text, press `Cmd/Ctrl + S`, continue typing (and I haven’t moved my hands from the keyboard at all). Without the extension, on the Write panel and using keyboard navigation, I’d need to press `tab` 3 times minimum (depending on OS and browser settings) to reach the Save button, then a minimum of 3 times `shift-option-tab` (Safari with FKA on)  to set the focus back on the main textarea.

**Comment by GugUser** [comment]
Hi Phil

Many thanks for your work.

The main function added by the rah_knots plugin isn't the position of the save button, rather the possibility to save articles, pages, forms etc. with the shortcut Ctrl + S or Cmd + S. I like that. In my own rah_knots version I removed the tooltip, I don't need it.

Sincerely, Claude.

**Comment by philwareham** [comment]
Ok, I support the adding of this functionality. @bloatware or @bloke can you look into this when you get a spare few minutes? I guess it would need to be consistent across all AJAX-Ified panels.

**Comment by bloatware** [comment]
I will have a very sporadic internet connection for yet another week, it's risky to develop this way, sorry. IMO, various js-enhancement should be part of theme, not everybody needs them. Or maybe @Bloke will find a magic way to plug them in/out?

**Comment by bloatware** [comment]
Please test. I wouldn't integrate textarea TAB into core, some people use it for page navigation.

**Comment by GugUser** [comment]
In my tests the shortcut Cmd + S (I'm on a Mac) for publish and save works fine, also for create or change sections, pages or forms. That is what I wanted. Good work, many thanks.

**Comment by bloatware** [comment]
Glad it works, please don't forget to close the issue if everything is fine.
